File: src/Domain/Entities/Ration.cs
namespace Domain.Entities;

public class Ration
{
    public int Id { get; set; }
    public required string Name { get; set; }
    
    public required int LivestockPropertyId { get; set; }
    public LivestockProperties LivestockProperties { get; set; } = null!;
    
    public ICollection<RationFeedType>? FeedTypes { get; set; }
}

File: src/WebApp/src/types/Ration.ts
import { RationFeedType } from './RationFeedType';

// Ration type
export interface Ration {
  id: number;
  name: string;
  rationTotalKg: number;
  rationTotalKgDs: number;
  dsProcentWeightedSum: number;
  averageTotalFeedTypes: Array<{ key: string; value: string |number }>

  feedTypes: RationFeedType[];
}
